src/features/klant/service.ts

```diff
@@ -209,61 +223,60 @@ export function useSearchKlanten<K extends KlantSearchField>({
   return ServiceResult.fromFetcher(getUrl, searchKlanten);
 }
 
-export function useKlantByBsn(
-  getBsn: () => string | undefined,
-): ServiceData<Klant | null> {
-  const getUrl = () => getKlantBsnUrl(getBsn());
-
-  return ServiceResult.fromFetcher(getUrl, searchSingleKlant, {
-    getUniqueId: () => getSingleBsnSearchId(getBsn()),
-  });
-}
-
 export function useUpdateContactGegevens() {
   return ServiceResult.fromSubmitter(updateContactgegevens);
 }
 
-export async function ensureKlantForBsn(
-  {
-    bsn,
-  }: {
-    bsn: string;
-  },
-  bronorganisatie: string,
-) {
-  const bsnUrl = getKlantBsnUrl(bsn);
-  const singleBsnId = getSingleBsnSearchId(bsn);
-
-  if (!bsnUrl || !singleBsnId) throw new Error();
+export async function ensureKlantForBsn({ bsn }: { bsn: string }) {
+  if (!bsn) throw new Error();
 
-  const first = await searchSingleKlant(bsnUrl);
+  const first = await getSingleKlantByBsn(bsn);
```

Comment:
het ophalen gebeurt in een aparte functie. het aanmaken gebeurt in deze functie. netter om dat uniformer te doen. 
dan zijn de paden ook wat overzichtelijker. nu is het soms een beetje zoeken hoe bijvoorbeeld de mapper gebruikt wordt en vanuit welke situaties en wat daar verschillende aan is. ik denk aan zoiets:


```
export async function ensureKlantForBsn({ bsn }: { bsn: string }) {
  if (!bsn) throw new Error();

  const partij = await getSingleKlantByBsn(bsn) ;

  if (!partij) {  
    partij = await createKlant(bsn);
  }

  klant = await mapPartijToKlant(partij);
  refreshCache(klant)

  return klant;
}
```

Reply:
Ik heb de functies iets anders opgesplitst en het komt nu in de buurt van de code in je comment.